### PR TITLE
chore: Generate `mongodbatlas_stream_privatelink_endpoint` file structure and resource schema

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Generate `mongodbatlas_stream_privatelink_endpoint` file structure and resource schema

Link to any related issue(s): CLOUDP-289116

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/streamprivatelinkendpoint/resource_schema.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+				MarkdownDescription: "The ID of the Private Link connection.",
+			},
+			"dns_domain": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Domain name of Privatelink connected cluster.",
+			},
+			"dns_sub_domain": schema.ListAttribute{
+				Optional:            true,
+				MarkdownDescription: "Sub-Domain name of Confluent cluster. These are typically your availability zones.",
+				ElementType:         types.StringType,
```
Comment (AgustinBettati) [inline internal/service/streamprivatelinkendpoint/resource_schema.go:24]: curious is this an actual array in the API? Just want to check that the Open API is correct.

File: tools/codegen/config.yml [hunk 2]
```diff
@@ -65,3 +65,16 @@ resources:
     schema:
       aliases:
         group_id: project_id
+
+  stream_privatelink_endpoint:
```
Comment (AgustinBettati) [inline tools/codegen/config.yml:69]: curious if any manual follow up change were needed 😄

File: internal/service/streamprivatelinkendpoint/resource.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+const resourceName = "stream_privatelink_endpoint"
+
+var _ resource.ResourceWithConfigure = &rs{}
+var _ resource.ResourceWithImportState = &rs{}
+
+func Resource() resource.Resource {
```
Comment (AgustinBettati) [inline internal/service/streamprivatelinkendpoint/resource.go:17]: you can probably add in the provider as part of this PR

**Review by AgustinBettati — APPROVED** [review]

File: internal/service/streamprivatelinkendpoint/resource_schema.go (hunk 1 above)
Comment (oarbusi) [reply internal/service/streamprivatelinkendpoint/resource_schema.go:24]: Yes it is an array, see the [API docs](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Streams/operation/createPrivateLinkConnection)

File: internal/service/streamprivatelinkendpoint/resource.go (hunk 3 above)
Comment (oarbusi) [reply internal/service/streamprivatelinkendpoint/resource.go:17]: As a personal preference, I intend to do this when I implement the resource

File: tools/codegen/config.yml (hunk 2 above)
Comment (oarbusi) [reply tools/codegen/config.yml:69]: I had to adjust the attributes from `Optional: true` to either `Required: true` or `Computed: true`. These were the only manual adjustments needed

**Pull request merged**